src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = Object.assign({}, state);
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      const stateKeys = Object.keys(newState);
+
+      for (const key of stateKeys) {
+        delete newState[key];
+      }
+
+      result.push({ ...newState });
+    } else if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete newState[key];
+      }
+
+      result.push({ ...newState });
+    } else if (action.type === 'addProperties') {
+      Object.assign(newState, action.extraData);
+
+      result.push({ ...newState });
+    }
```

Comment:
You should use switch construction here. And you can make result.push only once at the end of all if statements (at the end of all cases in case of switch).